lib/src/layer/tile_layer.dart

```diff
@@ -347,14 +352,26 @@ class _TileLayerState extends State<TileLayer> with TickerProviderStateMixin {
       for (var tile in tilesToRender) _createTileWidget(tile)
     ];
 
+    var content = options.colorFilter == null
+        ? Container(
+            color: options.backgroundColor,
+            child: Stack(
+              children: tileWidgets,
+            ),
+          )
+        : ColorFiltered(
+            colorFilter: options.colorFilter,
+            child: Container(
+              color: options.backgroundColor,
+              child: Stack(
+                children: tileWidgets,
+              ),
+            ),
+          );
+
```

Comment:
I think you can de-duplicate code by doing:
```suggestion
  var content = Container(
    color: options.backgroundColor,
    child: Stack(
      children: tileWidgets,
    ),
  );
  if (options.colorFilter == null) {
    content = ColorFiltered(colorFilter: options.colorFilter, child: content);
  }
```

Reply:
Yes, okay resolved.

Reply:
Okay, with a cast it works.
```
...
 content = ColorFiltered(colorFilter: options.colorFilter, child: content) as Container;
...
````

Reply:
I had to a a cast to a `Container` for compilation.
```
    if (options.colorFilter != null) {
      content = ColorFiltered(colorFilter: options.colorFilter, child: content) as Container;
    }
```
But then there is a runtime exception because "*ColorFiltered* is not a subtype of type *Container*".

Reply:
Explicit typedef would do the trick:
```
    Widget content = Container(
      color: options.backgroundColor,
      child: Stack(
        children: tileWidgets,
      ),
    );
    if (options.colorFilter != null) { // this should be != null
      content = ColorFiltered(colorFilter: options.colorFilter, child: content);
    }
```

Reply:
Yes, it works now.